test/test-applications/integrations/Samples.MultiDomainHost.Runner/Program.cs

```diff
@@ -90,6 +92,13 @@ private static void CreateAndRunAppDomain(string appName)
                 Console.WriteLine($"Finished code execution in AppDomain {appName}");
                 Console.WriteLine("**********************************************");
                 Console.WriteLine();
+
+                if (domain != null)
+                {
+                    // Wait for traces to be flushed, then unload the domain
+                    Thread.Sleep(2000);
```

Comment:
Not for this PR, but I'm thinking we should have a helper that calls `Tracer.Instance.FlushAsync()` using reflection in any sample apps that auto-close, given we know that's currently an oustanding issue?

Reply:
Here it's because we unload the appdomain (which causes threads to be aborted), but I don't think it should be needed elsewhere?

Reply:
> Here it's because we unload the appdomain (which causes threads to be aborted)
Makes sense

> but I don't think it should be needed elsewhere?
I'm not entirely sure. I thought we still needed @tonyredondo's fix in #1332 to be sure everything gets flushed (which we reverted in #1410). But a flaky test I thought was due to this looks like it's a separate issue, so maybe not